Eastern-Research-Group/expert-query
# Feature/302 clickable tooltips

## Related Issues:
* [EQ-302](https://jira.epa.gov/browse/EQ-302)

## Main Changes:
* Dropped the `@reach/tooltip` dependency, and switched to a version of the USWDS tooltip adapted for React.
  * This version draws from the USWDS source code as well as another adaptation from [uswds-react](https://github.com/trussworks/react-uswds/tree/main/src/components/Tooltip).
* Fixed a field label accessibility issue, where the name presented to a screen reader included tooltip and input content.

## Steps To Test:
1. Test the tooltips on a variety of screen sizes. Verify that touching the tooltip trigger on a touchscreen device or emulator opens the tooltip, and that clicking the trigger on desktop does not close the tooltip. Additionally, focusing the trigger with the keyboard should open the tooltip.

## N.B.
When testing the tooltips in the browser's mobile emulator at <= 360px, I found that some long tooltips (e.g. _Associated Action ID (Name))_ could not be positioned properly, and a warning is printed to the console in this case. However, the same tooltip displays fine on my iPhone and on a very narrow desktop window, so I believe the issue is related to the calculation of `win.innerHeight` or `document.documentElement.clientHeight` when using the emulator, and can be disregarded.
